collider/CMakeLists.txt

```diff
@@ -1,29 +1,15 @@
 project (collider)
 
-add_library(${PROJECT_NAME} src/collider.cpp)
+file(GLOB_RECURSE SOURCE_FILES CONFIGURE_DEPENDS *.cpp)
+add_library(${PROJECT_NAME} ${SOURCE_FILES})
```

Comment:
Agreed this looks cleaner, but I avoided it after reading posts like this: https://stackoverflow.com/a/18538444

Reply:
But the second answer mentioned that

> In CMake 3.12, the [`file(GLOB ...)` and `file(GLOB_RECURSE ...)`](https://cmake.org/cmake/help/v3.12/command/file.html#glob) commands gained a `CONFIGURE_DEPENDS` option which reruns cmake if the glob's value changes.

which should not have the problem in the first answer. But I'm fine with reverting back to the original one if you prefer that.

Reply:
Oh, that's great to know. Okay, let's go with this then.